### PR TITLE
Next 39784/customer changes email and password

### PR DESCRIPTION
**Pull request by ocavli**
This PR migrates an existing Cypress test to Playwright.

After incorporating review feedback, the tests have been further detailed, and additional edge cases that were not covered in the previous version have been included. The updated tests and steps are as follows:

**As a customer, I must be able to change my email via account.**

- Register a valid account
- Attempt to change email to an invalid address
- Attempt to change email to the same address
- Change email to a new valid address
- Verify login with old email fails
- Verify login with new email

**As a customer, I must be able to change my password via account.**

- Register a new account
- Attempt to change password to an invalid (short) password
- Successfully change password to a valid password
- Verify login with old password fails
- Verify login with new password

### CODE REVIEW

File: tests/acceptance/tests/Account/CustomerCredentials.spec.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import { test } from '@fixtures/AcceptanceTest';
+
+test('As a customer, I must be able to change email via account.', { tag: '@Account' }, async ({
```
Comment (frobel) [inline tests/acceptance/tests/Account/CustomerCredentials.spec.ts]: What about the negativ test? So ein mean that it is not possible to change the email with a wrong email without a @

File: tests/acceptance/tests/Account/CustomerCredentials.spec.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    await StorefrontAccountLogin.logoutLink.click();
+    await ShopCustomer.expects(StorefrontAccountLogin.successAlert).toBeVisible();
+    await StorefrontAccountLogin.emailInput.fill(changedEmail.email);
+    await StorefrontAccountLogin.passwordInput.fill(customer.password);
+    await StorefrontAccountLogin.loginButton.click();
+    await ShopCustomer.expects(StorefrontAccount.personalDataCardTitle).toBeVisible();
+});
+
+
+test('As a customer, I must be able to change password via account.', { tag: '@Account' }, async ({
```
Comment (frobel) [inline tests/acceptance/tests/Account/CustomerCredentials.spec.ts]: What about a negativ test, so that we test what happens if the passwords are not match.

**Review by frobel — CHANGES_REQUESTED** [review]

**Review by vanpham-sw — COMMENTED** [review]
I would change the **CustomerCredentials.spec.ts** file name to ensure that the new name accurately reflects the purpose and scope of the tests contained within the file. This will help maintain clarity and organization in our testing suite. 
CustomerChangeCredentials.spec.ts ?

File: tests/acceptance/tests/Account/CustomerCredentials.spec.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    await StorefrontAccountProfile.saveEmailAddressButton.click();
+    await ShopCustomer.expects(StorefrontAccountProfile.emailUpdateMessage).toBeVisible();
+    await ShopCustomer.expects(StorefrontAccountProfile.loginDataEmailAddress).toContainText(changedEmail.email);
+
+    await StorefrontAccountLogin.logoutLink.click();
+    await ShopCustomer.expects(StorefrontAccountLogin.successAlert).toBeVisible();
+    await StorefrontAccountLogin.emailInput.fill(changedEmail.email);
+    await StorefrontAccountLogin.passwordInput.fill(customer.password);
+    await StorefrontAccountLogin.loginButton.click();
+    await ShopCustomer.expects(StorefrontAccount.personalDataCardTitle).toBeVisible();
```
Comment (vanpham-sw) [inline tests/acceptance/tests/Account/CustomerCredentials.spec.ts]: To enhance the clarity and structure of your test script, you can break it down into logical steps using test.step. This approach helps others quickly understand the sequence of actions and the expected outcomes.

```suggestion
    await test.step('Navigate to login page and register a new customer', async () => {
    await ShopCustomer.goesTo(StorefrontAccountLogin.url());
    await ShopCustomer.attemptsTo(Register(customer));
    await ShopCustomer.expects(StorefrontAccount.page.getByText(customer.email, { exact: true })).toBeVisible();
});

await test.step('Change the registered customer\'s email address', async () => {
    await ShopCustomer.goesTo(StorefrontAccountProfile.url());
    await StorefrontAccountProfile.changeEmailButton.click();
    await ShopCustomer.expects(StorefrontAccountProfile.emailAddressInput).toBeVisible();
    await StorefrontAccountProfile.emailAddressInput.fill(changedEmail.email);
    await StorefrontAccountProfile.emailAddressConfirmInput.fill(changedEmail.email);
    await StorefrontAccountProfile.emailConfirmPasswordInput.fill(customer.password);
    await StorefrontAccountProfile.saveEmailAddressButton.click();
    await ShopCustomer.expects(StorefrontAccountProfile.emailUpdateMessage).toBeVisible();
    await ShopCustomer.expects(StorefrontAccountProfile.loginDataEmailAddress).toContainText(changedEmail.email);
});

await test.step('Log out and verify the logout success message', async () => {
    await StorefrontAccountLogin.logoutLink.click();
    await ShopCustomer.expects(StorefrontAccountLogin.successAlert).toBeVisible();
});

await test.step('Log in with the new email address', async () => {
    await StorefrontAccountLogin.emailInput.fill(changedEmail.email);
    await StorefrontAccountLogin.passwordInput.fill(customer.password);
    await StorefrontAccountLogin.loginButton.click();
    await ShopCustomer.expects(StorefrontAccount.personalDataCardTitle).toBeVisible();
});
```

File: tests/acceptance/tests/Account/CustomerCredentials.spec.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    await StorefrontAccountProfile.currentPasswordInput.fill(customer.password);
+    await StorefrontAccountProfile.saveNewPasswordButton.click();
+    await ShopCustomer.expects(StorefrontAccountProfile.passwordUpdateMessage).toBeVisible();
+
+    await StorefrontAccountLogin.logoutLink.click();
+    await ShopCustomer.expects(StorefrontAccountLogin.successAlert).toBeVisible();
+    await StorefrontAccountLogin.emailInput.fill(customer.email);
+    await StorefrontAccountLogin.passwordInput.fill(changedPassword.password);
+    await StorefrontAccountLogin.loginButton.click();
+    await ShopCustomer.expects(StorefrontAccount.personalDataCardTitle).toBeVisible();
```
Comment (vanpham-sw) [inline tests/acceptance/tests/Account/CustomerCredentials.spec.ts]: To enhance the readability and structure of test cases, adding test.step can be a great approach. It helps to break down the test into logical, understandable steps, making it easier for others to read and maintain. It'd also be great that, people will understand what the test case is doing in detail without understanding the code.

```suggestion
    await test.step('Register a new customer and verify registration', async () => {
    await ShopCustomer.goesTo(StorefrontAccountLogin.url());
    await ShopCustomer.attemptsTo(Register(customer));
    await ShopCustomer.expects(StorefrontAccount.page.getByText(customer.email, { exact: true })).toBeVisible();
});

await test.step('Change the customer\'s password', async () => {
    await ShopCustomer.goesTo(StorefrontAccountProfile.url());
    await StorefrontAccountProfile.changePasswordButton.click();
    await ShopCustomer.expects(StorefrontAccountProfile.newPasswordInput).toBeVisible();
    await StorefrontAccountProfile.newPasswordInput.fill(changedPassword.password);
    await StorefrontAccountProfile.newPasswordConfirmInput.fill(changedPassword.password);
    await StorefrontAccountProfile.currentPasswordInput.fill(customer.password);
    await StorefrontAccountProfile.saveNewPasswordButton.click();
    await ShopCustomer.expects(StorefrontAccountProfile.passwordUpdateMessage).toBeVisible();
});

await test.step('Log out and verify logout success message', async () => {
    await StorefrontAccountLogin.logoutLink.click();
    await ShopCustomer.expects(StorefrontAccountLogin.successAlert).toBeVisible();
});

await test.step('Log in with the new password', async () => {
    await StorefrontAccountLogin.emailInput.fill(customer.email);
    await StorefrontAccountLogin.passwordInput.fill(changedPassword.password);
    await StorefrontAccountLogin.loginButton.click();
    await ShopCustomer.expects(StorefrontAccount.personalDataCardTitle).toBeVisible();
});
```

File: tests/acceptance/tests/Account/CustomerCredentials.spec.ts (hunk 1 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/CustomerCredentials.spec.ts]: Added some edge cases 👍 

- invalid email without @
- already registered email
- try to log in with the old email after changing it

File: tests/acceptance/tests/Account/CustomerCredentials.spec.ts (hunk 2 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/CustomerCredentials.spec.ts]: Added some edge cases 👍 

- tried with short password (should be less the 8 chracters)
- tried to login with old password after changing it

**Comment by ocavli** [comment]
> I would change the **CustomerCredentials.spec.ts** file name to ensure that the new name accurately reflects the purpose and scope of the tests contained within the file. This will help maintain clarity and organization in our testing suite. CustomerChangeCredentials.spec.ts ?



> 

Good catch 👍 totally agree, already changed it as ChangeCustomerCredentials

File: tests/acceptance/tests/Account/CustomerCredentials.spec.ts (hunk 3 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/CustomerCredentials.spec.ts]: Added some steps 👍 Could you please check again

File: tests/acceptance/tests/Account/CustomerCredentials.spec.ts (hunk 4 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/CustomerCredentials.spec.ts]: Steps are added 👍

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    StorefrontAccount,
+    IdProvider,
+    Register,
+    StorefrontAccountProfile,
+
+}) => {
+
+    const customer = { email: IdProvider.getIdPair().uuid + '@test.com' , password: IdProvider.getIdPair().uuid };
+    const invalidEmail = 'invalidEmailWithoutAtSymbol';
+    const changedEmail = { email: IdProvider.getIdPair().uuid + '@test.com' };
```
Comment (yusufttur) [inline tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]:
```suggestion
    const newEmail = IdProvider.getIdPair().uuid + '@test.com';
```

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts [hunk 6]
```diff
@@ -0,0 +1,130 @@
+import { test } from '@fixtures/AcceptanceTest';
+
+test('As a customer, I must be able to change email via account.', { tag: '@Account' }, async ({
```
Comment (yusufttur) [inline tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]:
```suggestion
test('As a customer, I must be able to change my email via account.', { tag: '@Account' }, async ({
```

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        await StorefrontAccountProfile.changeEmailButton.click();
+        await ShopCustomer.expects(StorefrontAccountProfile.emailAddressInput).toBeVisible();
+        await StorefrontAccountProfile.emailAddressInput.fill(invalidEmail);
+        await StorefrontAccountProfile.emailAddressConfirmInput.fill(invalidEmail);
+        await StorefrontAccountProfile.emailConfirmPasswordInput.fill(customer.password);
+        await StorefrontAccountProfile.saveEmailAddressButton.click();
+        await ShopCustomer.expects(StorefrontAccountProfile.emailValidationAlert).toBeVisible();
+    });
+
+    await test.step('Attempt to change email to same address', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]:
```suggestion
    await test.step('Attempt to change email to the same address', async () => {
```

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+        await ShopCustomer.goesTo(StorefrontAccountProfile.url());
+        await StorefrontAccountProfile.changeEmailButton.click();
+        await StorefrontAccountProfile.emailAddressInput.fill(customer.email);
+        await StorefrontAccountProfile.emailAddressConfirmInput.fill(customer.email);
+        await StorefrontAccountProfile.emailConfirmPasswordInput.fill(customer.password);
+        await StorefrontAccountProfile.saveEmailAddressButton.click();
+        await ShopCustomer.expects(StorefrontAccountProfile.emailUpdateFailureAlert).toBeVisible();
+    });
+
+    await test.step('Attempt to change email to a new valid address', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]:
```suggestion
    await test.step('Change email to a new valid address', async () => {
```

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+        await ShopCustomer.goesTo(StorefrontAccountProfile.url());
+        await StorefrontAccountProfile.changeEmailButton.click();
+        await StorefrontAccountProfile.emailAddressInput.fill(changedEmail.email);
+        await StorefrontAccountProfile.emailAddressConfirmInput.fill(changedEmail.email);
```
Comment (yusufttur) [inline tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]:
```suggestion
        await StorefrontAccountProfile.emailAddressInput.fill(newEmail);
        await StorefrontAccountProfile.emailAddressConfirmInput.fill(newEmail);
```

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+        await StorefrontAccountProfile.emailConfirmPasswordInput.fill(customer.password);
+        await StorefrontAccountProfile.saveEmailAddressButton.click();
+        await ShopCustomer.expects(StorefrontAccountProfile.emailUpdateMessage).toBeVisible();
+        await ShopCustomer.expects(StorefrontAccountProfile.loginDataEmailAddress).toContainText(changedEmail.email);
```
Comment (yusufttur) [inline tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]:
```suggestion
        await ShopCustomer.expects(StorefrontAccountProfile.loginDataEmailAddress).toContainText(newEmail);
```

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+    });
+
+    await test.step('Try to login with the old email address', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]:
```suggestion
    await test.step('Verify login with old email fails', async () => {
```

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+        await StorefrontAccountLogin.logoutLink.click();
+        await ShopCustomer.expects(StorefrontAccountLogin.successAlert).toBeVisible();
+        await StorefrontAccountLogin.emailInput.fill(customer.email);
+        await StorefrontAccountLogin.passwordInput.fill(customer.password);
+        await StorefrontAccountLogin.loginButton.click();
+        await ShopCustomer.expects(StorefrontAccountLogin.invalidCredentialsAlert).toBeVisible();
+    });
+    
+    await test.step('Verify login with the new email address', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]:
```suggestion
    await test.step('Verify login with new email', async () => {
```

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+        await StorefrontAccountLogin.emailInput.fill(changedEmail.email);
```
Comment (yusufttur) [inline tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]:
```suggestion
        await StorefrontAccountLogin.emailInput.fill(newEmail);
```

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+        await StorefrontAccountLogin.passwordInput.fill(customer.password);
+        await StorefrontAccountLogin.loginButton.click();
+        await ShopCustomer.expects(StorefrontAccount.personalDataCardTitle).toBeVisible();
+    });
+});
+
+test('As a customer, I must be able to change password via account.', { tag: '@Account' }, async ({
```
Comment (yusufttur) [inline tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]:
```suggestion
test('As a customer, I must be able to change my password via account.', { tag: '@Account' }, async ({
```

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts [hunk 15]
```diff
@@ -0,0 +1,130 @@
...
+    StorefrontAccount,
+    IdProvider,
+    Register,
+    StorefrontAccountProfile,
+
+}) => {
+
+    const customer = { email: IdProvider.getIdPair().uuid + '@test.com' , password: IdProvider.getIdPair().uuid };
+    const invalidPassword = { password: 'short' }; // Invalid: less than 8 characters
+    const changedPassword = { password: IdProvider.getIdPair().uuid };
```
Comment (yusufttur) [inline tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]:
```suggestion
    const newPassword = IdProvider.getIdPair().uuid;
```

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts [hunk 16]
```diff
@@ -0,0 +1,130 @@
...
+        await StorefrontAccountProfile.changePasswordButton.click();
+        await ShopCustomer.expects(StorefrontAccountProfile.newPasswordInput).toBeVisible();
+        await StorefrontAccountProfile.newPasswordInput.fill(changedPassword.password);
+        await StorefrontAccountProfile.newPasswordConfirmInput.fill(changedPassword.password);
+        await StorefrontAccountProfile.currentPasswordInput.fill(customer.password);
+        await StorefrontAccountProfile.saveNewPasswordButton.click();
+        await ShopCustomer.expects(StorefrontAccountProfile.passwordUpdateMessage).toBeVisible();
+    });
+
+    await test.step('Try to login with old password', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]:
```suggestion
    await test.step('Verify login with old password fails', async () => {
```

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts [hunk 17]
```diff
@@ -0,0 +1,130 @@
...
+        await StorefrontAccountLogin.logoutLink.click();
+        await ShopCustomer.expects(StorefrontAccountLogin.successAlert).toBeVisible();
+        await StorefrontAccountLogin.emailInput.fill(customer.email);
+        await StorefrontAccountLogin.passwordInput.fill(customer.password);
+        await StorefrontAccountLogin.loginButton.click();
+        await ShopCustomer.expects(StorefrontAccountLogin.invalidCredentialsAlert).toBeVisible();
+    });
+
+    await test.step('Login with changed password to verify', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]:
```suggestion
    await test.step('Verify login with new password', async () => {
```

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts [hunk 18]
```diff
@@ -0,0 +1,130 @@
...
+        await ShopCustomer.goesTo(StorefrontAccountLogin.url());
+        await StorefrontAccountLogin.emailInput.fill(customer.email);
+        await StorefrontAccountLogin.passwordInput.fill(changedPassword.password);
```
Comment (yusufttur) [inline tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]:
```suggestion
        await StorefrontAccountLogin.passwordInput.fill(newPassword);
```

**Review by yusufttur — CHANGES_REQUESTED** [review]
I suggested some improvements to the description and test steps.

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts (hunk 5 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]: Fixed 👍

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts (hunk 6 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]: Fixed 👍

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts (hunk 7 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]: Fixed 👍

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts (hunk 8 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]: Fixed 👍

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts (hunk 10 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]: Fixed 👍

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts (hunk 11 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]: Fixed 👍

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts (hunk 12 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]: Fixed 👍

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts (hunk 13 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]: Fixed 👍

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts (hunk 15 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]: Fixed 👍

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts (hunk 16 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]: Fixed 👍

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts (hunk 17 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]: Fixed 👍

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts (hunk 18 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]: Fixed 👍

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts (hunk 14 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]: Fixed 👍

File: tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts (hunk 9 above)
Comment (ocavli) [reply tests/acceptance/tests/Account/ChangeCustomerCredentials.spec.ts]: Fixed 👍

**Review by frobel — APPROVED** [review]

**Review by yusufttur — APPROVED** [review]

**Review by vanpham-sw — APPROVED** [review]
